app/controllers/downloads_controller.rb

```diff
@@ -21,7 +21,7 @@ def check_download_params
   end
 
   def downloader
-    @downloader ||= Rails.configuration.x.storage_adapter.constantize::Downloader.new(key: key)
+    @downloader ||= Storage::S3::Downloader.new(key: key)
```

Comment:
This is the major change. Always use S3 rather than the option of a local storage adapter when run in test/dev.

Reply:
I'm always weary of memoizing stateful objects, have seen them become polluted.
Fine for now though